starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -695,63 +686,52 @@ sub check_dither {
-    my $or_val;
-    if (defined $self->{DITHER_ON}){
-        $or_val = $dthr_cmd{$self->{DITHER_ON}};
-        # ACA-002
-        push @{$self->{warn}}, "$alarm Dither mismatch - OR: $or_val != Backstop: $bs_val\n"
-            if ($or_val ne $bs_val);
-    }
-    else{
-        push @{$self->{warn}},
-            "$alarm Unable to determine dither from OR list\n";
-    }
-
-    # If dither is anabled according to the OR, check that parameters match OR vs Backstop
-    if ((defined $or_val) and ($or_val eq 'ENAB')){
-        my $y_amp = $self->{DITHER_Y_AMP} * 3600;
-        my $z_amp = $self->{DITHER_Z_AMP} * 3600;
-        if (defined $dither->{ampl_y}
-                and defined $dither->{ampl_p}
-                    and (abs($y_amp - $dither->{ampl_y}) > 0.1
-                             or abs($z_amp - $dither->{ampl_p}) > 0.1)){
-            my $warn = sprintf("$alarm Dither amp. mismatch - OR: (Y %.1f, Z %.1f) "
-                                   . "!= Backstop: (Y %.1f, Z %.1f)\n",
-                               $y_amp, $z_amp,
-                               $dither->{ampl_y}, $dither->{ampl_p});
-            push @{$self->{warn}}, $warn;
+    # Determine dither at acquisition
+    my $acq_dither;
+    foreach my $dither_state (reverse @{$dthr}) {
+	if ($obs_tstart >= $dither_state->{time}) {
+            $acq_dither = $dither_state;
+            last;
         }
     }
-    # Check for standard and large dither based solely on backstop/history values
-    if (($bs_val eq 'ENAB') and (defined $dither->{ampl_y} and defined $dither->{ampl_p})){
-        $self->{cmd_dither_y_amp} = $dither->{ampl_y};
-        $self->{cmd_dither_z_amp} = $dither->{ampl_p};
-        if (not standard_dither($dither)){
+
+
+    $self->{dither_acq} = $acq_dither;
+    $self->{dither_guide} = $guide_dither;
+
+    # Check for standard and large dither
+    # This needs to be before the checking for dither during the observation, because the pad
+    # for that checking is set here
+    if ($guide_dither->{state} eq 'ENAB'){
+        if ((not standard_dither($guide_dither)) or not standard_dither($acq_dither)){
             push @{$self->{yellow_warn}}, "$alarm Non-standard dither\n";
-            if ($dither->{ampl_y} > $large_dith_thresh or $dither->{ampl_p} > $large_dith_thresh){
-                $self->large_dither_checks($dither, $dthr);
+            if ($guide_dither->{ampl_y} > $large_dith_thresh or $guide_dither->{ampl_p} > $large_dith_thresh){
+                $self->large_dither_checks($guide_dither, $dthr);
                 # If this is a large dither, set a larger pad at the end, as we expect
                 # standard dither parameters to be commanded at 5 minutes before end,
                 # which is greater than the 3 minutes used in the "no dither changes
                 # during observation check below
                 $obs_end_pad = 5.5 * 60;
             }
         }
+        else{
```

Comment:
Likewise move this test out of the `else`.  Sure, given the current definition of `standard_dither()` this test cannot fail for both guide and acq dither being "standard", but it just makes the code logic more difficult to follow and more fragile (what if 20x20 becomes a standard dither for ACIS?  Could happen to support dynamic background).